### ISSUE TITLE
Support Google provider 4.0 for Workload Identity

### ISSUE THREAD

**Issue by jackwhelpton** [terraform-google-modules/terraform-google-kubernetes-engine#1068]
### TL;DR

Because the WI module constrains the Google provider to `version ~> 3.0`, it can't be used in conjunction with other modules that require `version >= 4.0`.

### Expected behavior

Module is usable with Google provider 4.0.

### Observed behavior

```sh
Could not retrieve the list of available versions for provider hashicorp/google: no available releases match the
│ given constraints >= 2.12.0, >= 3.43.0, >= 3.45.0, ~> 3.53, >= 4.0.0, < 4.0.0, < 5.0.0
```

### Terraform Version

```sh
Terraform v1.0.11
on windows_amd64
+ provider registry.terraform.io/hashicorp/external v2.1.0
+ provider registry.terraform.io/hashicorp/google v3.90.0
+ provider registry.terraform.io/hashicorp/google-beta v3.90.0
+ provider registry.terraform.io/hashicorp/helm v2.3.0
+ provider registry.terraform.io/hashicorp/kubernetes v2.6.1
+ provider registry.terraform.io/hashicorp/null v3.1.0
+ provider registry.terraform.io/hashicorp/random v3.1.0
+ provider registry.terraform.io/netapp/netapp-gcp v21.5.0
```

### Additional information

There's a PR raised for this by an automated process (https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/pull/1066) but it needs some work.

I don't mind taking a run at this, as I've made similar changes to our internal modules.. we need to remove the basic auth variables/settings and make some other changes to field names: it'd probably be a breaking change, unless we keep the variables and just deprecate them.

That said, I'm not sure how my new submission would impact the existing PR: any guidance?

**Comment by jackwhelpton** [comment]
It looks like https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/pull/1052 is also related, althogh it attempts to address this issue without doing the provider upgrade (which I will need).

**Comment by PDQDakota** [comment]
I am also running into this issue using the 4.1.0 version of the hashicorp/google provider.

> Could not retrieve the list of available versions for provider hashicorp/google: no available releases match the given constraints >= 3.39.0, < 4.0.0, >= 4.1.0

**Comment by jackwhelpton** [comment]
Fwiw I'm having a stab at resolving this here: https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/pull/1071.
